---
title: "Ch_2"
author: "Amy Fox"
date: "2/13/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


Mixture Models

P(X | theta) = theta^x * e^-theta / x!

where x is the data and theta is the parameter

If I don't know what theta is, then I want find the value of theta that increases the probability to the highest amount


Poisson is often count data becuase it include non-negative data
```{r}
load("../data/e100.RData")
e99 = e100[-which.max(e100)]

barplot(table(e99), space = 0.8, col = "chartreuse4")


library("vcd")
gf1 = goodfit( e99, "poisson")
rootogram(gf1, xlab = "", rect_gp = gpar(fill = "chartreuse4"))
```

2.5.2
```{r}
library("Biostrings")
staph = readDNAStringSet("../data/staphsequence.ffn.txt", "fasta")

letterFrq = vapply(staph, letterFrequency, FUN.VALUE = numeric(4),
         letters = "ACGT", OR = 0)

colnames(letterFrq) = paste0("gene", seq(along = staph))
tab10 = letterFrq[, 1:10]
computeProportions = function(x) { x/sum(x) }
prop10 = apply(tab10, 2, computeProportions)
round(prop10, digits = 2)


p0 = rowMeans(prop10)
p0
```

The distrubtino of the nucleotide is not equal amongst genes. - Fig. 2.7  shows that the red is way outside the actual distribution (there is too much variability between genes)


# 2.6

Could jsut compared our observed to a Chi-squred because we know that our data is chi-squared

Chi takes on only  positive values with higher humper near 0


10 genes * 4 nucelotides - 1 (becuase when you know what 3 of the nucleotide frequencies the other one is 1-p)

```{r}
load("../data/ChargaffTable.RData")
ChargaffTable

# if these are related, then the (pC - pG)^2 shoud be small
statChf = function(x){
  sum((x[, "C"] - x[, "G"])^2 + (x[, "A"] - x[, "T"])^2)
}

chfstat = statChf(ChargaffTable)

permstat = replicate(100000, {
     permuted = t(apply(ChargaffTable, 1, sample))
     colnames(permuted) = colnames(ChargaffTable)
     statChf(permuted)
})

pChf = mean(permstat <= chfstat)
pChf
```


# Exercise 2.3

```{r}
mtb = read.table("../data/M_tuberculosis.txt", header = TRUE)
head(mtb, n = 4)

# proline occurs at this probability 
pro  =  mtb[ mtb$AmAcid == "Pro", "Number"]
pro/sum(pro)

# a)
# see that Arg, Leu, and Ser have a high number of codons at 6
rowSums(table(mtb))


# b)
# PerThous could be for exmaple that there are 19.25 cases of GGG per 1000 nucleotides

library(dplyr)

mtb %>%
  mutate(genome_length = sum(Number)) %>%
  group_by(Codon) %>%
  mutate(PerThous = 1000*Number / genome_length)

# c)
# Write an R function that you can apply to the table to find which of the amino acids shows the strongest codon bias, i.e., the strongest departure from uniform distribution among its possible spellings.
mtb

# see that Arg, Leu, and Ser have a high number of codons at 6
codon_no <- rowSums(table(mtb))
library(tidyverse)

codon_df <- data.frame(codon_no) %>%
  rownames_to_column() %>%
  rename(`AmAcid` = rowname) %>%
  mutate(prob_codon = 1/codon_no)

real_data <- full_join(mtb, codon_df, by = "AmAcid") %>%
  group_by(AmAcid) %>%
  mutate(am_length = sum(Number)) %>%
  mutate(real_prob = Number / am_length) %>%
  select(-codon_no, -am_length) %>%
  mutate(expected = Number*prob_codon)
  



# Something like this below???

stat = function(obsvd, exptd = 20 * pvec) {
   sum((obsvd - exptd)^2 / exptd)
}

B = 1000

simulstat = replicate(B, {
  randomtab10 = sapply(cs, function(s) { rmultinom(1, s, p0) })
  stat(randomtab10, expectedtab10)
})
S1 = stat(tab10, expectedtab10)
sum(simulstat >= S1)
```

*** # the number column is the observed, expected is the nubmer column times the prob_codon




# Chapter 2.9

We want to estimate the proportion of a particular Y-haplotype that consists of a set of different short tandem repeats (STR).

```{r}
haplo6=read.table("../data/haplotype6.txt",header = TRUE)
haplo6
```

We generate a random sample of 10000 Î¸'s

```{r}
rtheta = rbeta(100000, 50, 350)
y = vapply(rtheta, function(th) {
  rbinom(1, prob = th, size = 300)
}, numeric(1))
hist(y, breaks = 50, col = "orange", main = "", xlab = "")


# Question: does the following code give us the same result as above?
z <- rbinom(length(rtheta), rtheta, size = 300)
hist(z, breaks = 50, col = "orange", main = "", xlab = "")
```

Histogram of all the thetas such that Y=40: the posterior distribution
**where is thetas defined??**
```{r}
thetaPostEmp = rtheta[ y == 40 ]

hist(thetaPostEmp, breaks = 40, col = "chartreuse4", main = "",
  probability = TRUE, xlab = expression("posterior"~theta))
densPostTheory  =  dbeta(thetas, 90, 610)
lines(thetas, densPostTheory, type="l", lwd = 3)
```

Use Monte Carlo integration instead of above
```{r}
thetaPostMC = rbeta(n = 1e6, 90, 610)
mean(thetaPostMC)

qqplot(thetaPostMC, thetaPostEmp, type = "l", asp = 1)
abline(a = 0, b = 1, col = "blue")
```

# Chapter 2.10

Explore some of the useful data and functions provided in the Biostrings package by exploring the tutorial vignette.
```{r}
library("Biostrings")
library("BSgenome")
ag = available.genomes()
length(ag)
ag[1:2]
```

Explore the occurrence of the AGGAGGT motif3333 This is the Shine-Dalgarno motif which helps initiate protein synthesis in bacteria. in the geonome of E.coli. We use the genome sequence of one particular strain, Escherichia coli str. K12 substr.DH10B3434 It is known as the laboratory workhorse, often used in experiments., whose NCBI accession number is NC_010473.
```{r}
library("BSgenome.Ecoli.NCBI.20080805")
Ecoli
shineDalgarno = "AGGAGGT"
ecoli = Ecoli$NC_010473
```


Bayesian often considered "subjective" method 
Dicey working with Bayesian on drug trials 

If you have 4 strains that are different 
Have 2 strains that I know more about 

Can go back and think 

Don't have to use beta distrubtion for Bayesian - can use what ever is appropriate for what you think the data should look like

You always want to plot your prior and your data and and make sure that the posterior is not directly on top of your prior because that means that your pior is affecting you posterior completely/too much 


# Exercise 2.6
Choose your own prior for the parameters of the beta distribution. You can do this by sketching it here: https://jhubiostatistics.shinyapps.io/drawyourprior. Once you have set up a prior, re-analyse the data from Section 2.9.2, where we saw 
Y = 40 successes out of n = 300 trials. Compare your posterior distribution to the one we obtained in that section using a QQ-plot.

```{r}
set.seed(2)
alpha = 2.47
beta = 8.5

rtheta = rbeta(1000000, alpha, beta)
y = vapply(rtheta, function(th) {
  rbinom(1, prob = th, size = 300)
}, numeric(1))
hist(y, breaks = 50, col = "orange", main = "", xlab = "")


thetas = seq(0, 1, by = 0.001)

thetaPostEmp = rtheta[ y == 40 ]
hist(thetaPostEmp, breaks = 40, col = "chartreuse4", main = "",
  probability = TRUE, xlab = expression("posterior"~theta), ylim = c(0,35))
densPostTheory  =  dbeta(thetas, (alpha + 40), (beta + 300-40)) # what the heck is this
lines(thetas, densPostTheory, type="l", lwd = 3)


thetaPostMC = rbeta(n = 1e6, (alpha + 40), (beta + 300-40)) # this needs to be (alpha + 40) and (B + 300-40)
mean(thetaPostMC)

qqplot(thetaPostMC, thetaPostEmp, type = "l", asp = 1)
abline(a = 0, b = 1, col = "blue")
```

Good agreement between the two 

When theta (distribution of the probability)